spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/spel/MethodReferenceNode.java

```diff
@@ -63,13 +146,42 @@
 
 	/**
 	 * Returns the name of the method.
-	 * 
-	 * @return
+	 * @Deprecated since 1.10. Please use {@link #getMethodReference()}.
 	 */
+	@Deprecated
 	public String getMethodName() {
 
+		AggregationMethodReference methodReference = getMethodReference();
+		return methodReference != null ? methodReference.getMethodName() : null;
+	}
+
+	/**
+	 * Return the {@link AggregationMethodReference}.
+	 *
+	 * @return can be {@literal null}.
+	 * @since 1.10
+	 */
+	public AggregationMethodReference getMethodReference() {
+
 		String name = getName();
 		String methodName = name.substring(0, name.indexOf('('));
 		return FUNCTIONS.get(methodName);
 	}
+
+	/**
+	 * @author Christoph Strobl
+	 * @since 1.10
+	 */
+	@lombok.Data
+	@lombok.Builder
+	public static class AggregationMethodReference {
```

Comment:
How about an custom written builder that is not `public` and is tailored to the construction flow (i.e. method name always required, adoption of argument type characteristics)?